### PR TITLE
Core/Script Paladin Argent Defender

### LINKED ISSUE #245: Talent Paladin tank [Ardent Defender] not works
**Current Behavior**: The talent of paladin [Ardent Defender] does not activate when you have less than 1% health

**Expected behavior**: The talent [Ardent Defender] should be activated, heal up to 30% health and leave you a debuff of 2 minutes so that the talent does not return activating.

**Link**: http://www.wowhead.com/spell=31850/ardent-defender

**Branch(es)**: 1.x

**AC hash/commit**:  https://github.com/azerothcore/azerothcore-wotlk/commit/cd6209cdb10234201682fd4c15b01964c0fa60ed

**Operating system**:  Windows 2008 R2

**Modules**:  Are you using modules? if yes, list them

**Other Customizations**: 
- Transmogrif
- Crossfaction

### PR DESCRIPTION
**Pull request by talamortis**
**Changes proposed:**
Fix Spell for Paladin Argent Defender

**Target branch(es):** Master

**Issues addressed:** Closes #245 

**Tests performed:** (Does it build, tested in-game, etc)
Build succeeded & Tested in game.

**Known issues and TODO list:**

None

**NOTE** You no longer need to squash your commits, on merge we will squash it for you. (GitHub added a new feature)

Fix Spell Paladin Argent Defender

### CODE REVIEW

File: src/scripts/Spells/spell_paladin.cpp [hunk 1]
```diff
@@ -336,7 +336,7 @@ class spell_pal_ardent_defender : public SpellScriptLoader
                     // Max heal when defense skill denies critical hits from raid bosses
                     // Formula: max defense at level + 140 (raiting from gear)
                     uint32 reqDefForMaxHeal  = victim->getLevel() * 5 + 140;
-                    float pctFromDefense = (defenseSkillValue - victim->getLevel()*5.0f) / 140.0f;
+                    float pctFromDefense = (defenseSkillValue = victim->getLevel()*5.0f) / 140.0f;
```
Comment (ShinDarth) [inline src/scripts/Spells/spell_paladin.cpp]: to be honest I don't get the aim of this. Why are you assigning ``victim->getLevel()*5.0f`` to  ``defenseSkillValue`` if this variable will not be used anywhere ?

**Comment by HeartWell** [comment]
https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Spells/spell_paladin.cpp#L189

File: src/scripts/Spells/spell_paladin.cpp [hunk 2]
```diff
@@ -336,7 +336,7 @@ class spell_pal_ardent_defender : public SpellScriptLoader
                     // Max heal when defense skill denies critical hits from raid bosses
                     // Formula: max defense at level + 140 (raiting from gear)
                     uint32 reqDefForMaxHeal  = victim->getLevel() * 5 + 140;
-                    float pctFromDefense = (defenseSkillValue - victim->getLevel()*5.0f) / 140.0f;
+                    float pctFromDefense = (defenseSkillValue >= reqDefForMaxHeal);
```
Comment (ShinDarth) [inline src/scripts/Spells/spell_paladin.cpp]: this is wrong:  the ``>=`` is a logical operator and returns a boolean value, which you are trying to assign to a float.

**Comment by ShinDarth** [comment]
also please open one separate PR for each separate thing

**Comment by sucofog** [comment]
News?

**Pull request closed without merging**